src/Kiota.Builder/Refiners/PhpRefiner.cs

```diff
@@ -0,0 +1,86 @@
+            ReplaceIndexersByMethodsWithParameter(generatedCode, generatedCode, false, "ById");
+            AddPropertiesAndMethodTypesImports(generatedCode, true, false, true);
+            AddGetterAndSetterMethods(generatedCode,new HashSet<CodePropertyKind>()
+            {
+                CodePropertyKind.Custom,
+                CodePropertyKind.AdditionalData,
+                CodePropertyKind.BackingStore
+            }, _configuration.UsesBackingStore, true);
+            ReplaceReservedNames(generatedCode, new PhpReservedNamesProvider(), reservedWord => $"Graph{reservedWord.ToFirstCharacterUpperCase()}");
+            AddParsableInheritanceForModelClasses(generatedCode);
+            ReplaceBinaryByNativeType(generatedCode, "StreamInterface", "Psr\\Http\\Message", true);
+            MoveClassesWithNamespaceNamesUnderNamespace(generatedCode);
+            CorrectCoreType(generatedCode, null, CorrectPropertyType);
+           
+        }
+        
+        private static readonly AdditionalUsingEvaluator[] defaultUsingEvaluators = { 
+            new (x => x is CodeProperty prop && prop.IsOfKind(CodePropertyKind.RequestAdapter),
+                "Microsoft\\Kiota\\Abstractions", "RequestAdapter"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.RequestGenerator),
+                "Microsoft\\Kiota\\Abstractions", "HttpMethod", "RequestInformation", "MiddlewareOption"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.RequestExecutor),
+                "Microsoft\\Kiota\\Abstractions", "ResponseHandler"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.Serializer),
+                "Microsoft\\Kiota\\Abstractions\\Serialization", "SerializationWriter"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.Deserializer),
+                "Microsoft\\Kiota\\Abstractions\\Serialization", "ParseNode"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.RequestExecutor),
+                "Microsoft\\Kiota\\Abstractions\\Serialization", "Parsable"),
+            new (x => x is CodeClass @class && @class.IsOfKind(CodeClassKind.Model),
+                "Microsoft\\Kiota\\Abstractions\\Serialization", "Parsable"),
+            new (x => x is CodeMethod method && method.IsOfKind(CodeMethodKind.ClientConstructor) &&
+                      method.Parameters.Any(y => y.IsOfKind(CodeParameterKind.BackingStore)),
+                "Microsoft\\Kiota\\Abstractions\\Store", "BackingStoreFactory", "BackingStoreFactorySingleton"),
+            new (x => x is CodeProperty prop && prop.IsOfKind(CodePropertyKind.BackingStore),
+                "Microsoft\\Kiota\\Abstractions\\Store", "BackingStore", "BackedModel", "BackingStoreFactorySingleton" ),
+        };
+        
+        private static void AddParsableInheritanceForModelClasses(CodeElement currentElement) {
```

Comment:
instead of duplicating, move to the common language, set to protected, and remove the copy in java

Reply:
We probably do a bit better and move it up for C#, TypeScript and Ruby as well and pass in the `Name` for the declaration as a parameter.

Reply:
Hopefully the fix implemented works for you @andrueastman, @baywet.